### DISCUSSION TITLE
Docker Swarm (existing)

### DISCUSSION THREAD

**Discussion by prologic** [Dokploy/dokploy#62] [Q&A]
What's the support and experience like setting this up with an existing Docker Swarm cluster?

I already have:

- 10 node Docker Swarm Cluster
- Traefik for ingress
- Numerous applications and services already deployed

**Comment by TheoCouss** [comment]
Given the extensive number of nodes and applications you currently manage, I would advise against replacing your existing configurations with Dokploy at this time. The software is still in its early development stages, and there appears to be no clear roadmap for the project. Furthermore, migrating your applications to Dokploy would require handling each one individually. Additionally, it seems Dokploy does not yet support setting constraints, managing replicas, etc.

**Reply by Siumauricio** [reply]
That's right it uses dokploy under the hood it uses Docker swarm, the next feature request we are working on is support for multiple nodes, with that it would be very easy to implement replicas and such.

Regarding the roadmap, I already answered about what we will work on in the near future, I want this to be something stable and easy to use and I think it is something that many people are currently appreciating, there have been very few bugs that have already been resolved, so we could focus on that priority for now, then the implementation of docker-compose will surely come

**Comment by Siumauricio** [comment]
Hi @prologic,

Currently there is no exact or specific way to migrate projects where they are currently running, I could not give you an answer, my only suggestion would be to migrate the projects one by one, since dokploy is currently designed to handle it through its UI, no from outside.


Some people have asked similar questions about how to migrate current containers that were not created by dokploy, but it is something we can investigate to be able to import or automatically detect these containers in the future.

**Reply by prologic** [reply]
My question isn't specify about migration, but running Dockploy alongside an already existing setup.

**Reply by Siumauricio** [reply]
@prologic I added a simple instruction for existing docker swarm 

https://docs.dokploy.com/en/docs/core/get-started/manual-installation#existing-docker-swarm
